.github/workflows/check-docs-pr-preview.yml

```diff
@@ -1,64 +1,87 @@
+          clean: false
+          single-commit: true
+
+      - name: Update PR with Preview Link
+        uses: actions/github-script@v6
+        if: success()
+        with:
+          script: |
+            const previewUrl = `https://${context.repo.owner}.github.io/${context.repo.repo}/preview/pr-${context.payload.pull_request.number}/`;
+            const commentBody = `🌐 Documentation preview automatically generated:\n${previewUrl}`;
+            
+            // Check if bot has already commented
+            const comments = await github.rest.issues.listComments({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: context.payload.pull_request.number
+            });
+            
+            const botComment = comments.data.find(comment => 
+              comment.user.login === 'github-actions[bot]' && 
+              comment.body.includes('Documentation preview')
+            );
+            
+            if (!botComment) {
+              await github.rest.issues.createComment({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                issue_number: context.payload.pull_request.number,
+                body: commentBody
+              });
+            }
 
-          if ! PREVIEW_LINE=$(echo "$BODY" | grep -iE '^\s*Preview.*https?://') ; then
-            echo "${RED}[ERROR] Docs PR: No preview link found. Please include a line starting with 'Preview' and a valid GitHub Pages URL for your changes.${NC}"
-            echo
-            exit 1
-          else
-            if [[ "$PREVIEW_LINE" == *"$GITHUB_PAGES_URL"* ]]; then
-              if [[ "$BRANCH" =~ $DOC_BRANCH_REGEX ]]; then
-                echo "${GREEN}[OK] Docs PR: doc- branch and preview link format matches expected GitHub Pages URL.${NC}"
-              else
-                echo "${YELLOW}[WARN] Docs PR: Preview link format is good, but branch does not start with doc-. Consider renaming to enable automatic previews.${NC}"
-              fi
-            else
-              echo "${YELLOW}[WARN] Docs PR: Preview link found, but it does not match the expected GitHub Pages URL."
-              PREVIEW_URL=$(echo "$PREVIEW_LINE" | grep -oE 'https?://[^ ]*' || true)
-              echo "Expected: ${GITHUB_PAGES_URL}"
-              echo "Found:    ${PREVIEW_URL}${NC}"
-            fi
-          fi
-          echo
+            // Update PR description if needed
+            const prBody = context.payload.pull_request.body || '';
+            if (!prBody.includes(previewUrl)) {
+              const newBody = prBody + `\n\nPreview: ${previewUrl}`;
+              await github.rest.pulls.update({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                pull_number: context.payload.pull_request.number,
+                body: newBody
+              });
+            }
```

Comment:
GitHub is displaying an alarming symbol here:
<img width="129" height="82" alt="image" src="https://github.com/user-attachments/assets/dfe21e80-9559-4bf2-a795-5bdbcd8f493b" />